### DISCUSSION TITLE
Open/unpack the file on Linux

### DISCUSSION THREAD

**Discussion by sakkamade** [gildas-lormeau/SingleFileZ#94] [Q&A] [answered]
Good day. 

Thank you for amazing extension. _This_ one, unlike SingleFile, appears to be precisely what I am looking for.
And yet, the issue came upon me the the very minute I installed it: I can't unpack it.

The application I am using is Ark which itself depends upon: p7zip library (for 7Z format), unrar library (for RAR decompression support), unarchiver (for RAR format support) and some other libraries, however I have also tired the program _xarchiver_, which has support of some additional libraries. 
And I can positively confirm that with each and every of the zip-libraries installed, I am unable to unzip, nor to even open, the generated `.html` archive as long as I left option "create self extracting archives" enabled. When I disable it, I am able to plainly open and unpack the archive.

Yes, I have no knowledge upon archives structure, most likely the issue is not a concern of the SingleFIleZ itself, and I will undoubtedly proceed searching for the way to make it work, still, if I discover the way to open the file, I name it here, and this would certainly help others in future, otherwise, supposing someone is already learned the way, they may help me at present.

**Comment by gildas-lormeau** [comment, marked as answer]
You should be able to unzip self-extracting archives with the `unzip` command.

**Reply by sakkamade** [reply]
Yeah.

**Reply by sakkamade** [reply]
Thank you!

**Comment by sakkamade** [comment]
Well, spending some more time on search, now, I found no duly GUI solution still. However, there is a `p7zip-gui`, and one might give it a try (I didn't).

One way could have been to create a `.desktop` file with `Exec=unzip %U -d *exdir*`, however the `%N` (in place for `*exdir*`) was [deprecated](https://specifications.freedesktop.org/desktop-entry-spec/latest/apc.html), although I am not sure if that would've worked, too. 
The only choice is to enable "create a root directory" in the extension's settings, then the command in `.desktop` file will simply be `Exec=unzip %U`.

One may try to come up with something similar by [unp](https://tracker.debian.org/pkg/unp) or [atool](https://www.nongnu.org/atool/) (I have read nothing of them).
Alternatively, simply running a command `unzip file[.zip] -d [exdir]` should do it.

Hence, the question is still open to solutions; though, at this point, I shall just create a bash script which will be called to by a `.desktop` file.

Curious matter is that the _xarchive_ availing itself of an `unzip` package and fails to open the archive all the same. 
None the less, I won't go in depths so to learn why.

**Comment by sakkamade** [comment]
Here is some GUI-ish way (`zenity` is required):
```desktop
[Desktop Entry]
Comment=Extract the zip-HTML file to specific location
Exec=unzip -n %f -d "$(zenity --file-selection --directory)"/%n-unzipped
Icon=archive
MimeType=text/html;
Name=Unzip HTML
Path=
StartupNotify=true
Terminal=false
Categories=Utilities;
```
This will extract the archive into the "`filename.html`-unzipped" directory of the specified location, and will never overwrite existing files (although some files may still be extracted).

**Reply by sakkamade** [reply]
> however the `%N` (in place for `*exdir*`) was [deprecated](https://specifications.freedesktop.org/desktop-entry-spec/latest/apc.html),

Ha! Although deprecated, it is not removed yet. So this works too:

```desktop
Exec=unzip -n %U -d "$(zenity --file-selection --directory)"/%N-unzipped
```

(will update above as well)

**Reply by sakkamade** [reply]
Also, simply this will unpack the file to the same location as archive, into a sub-folder:
```desktop
Exec=unzip -n %f -d %n-unzipped
```

**Comment by gildas-lormeau** [comment]
I forgot to mention that you can also simply disable the option `ZIP content > create self extracting archives` if you don't need this feature. Thus, SingleFileZ will produce zip files compatible with any unzip tool.
